figures/S5.supplemental_figure5/scripts/S5.r

```diff
@@ -0,0 +1,1448 @@
+# set to string
+filename <- filename[[1]]
+il6_final <- read.csv(filename, header = TRUE, sep = ",", stringsAsFactors = FALSE)
+
+il6_final <- process_subset_data(il6_final)
+head(il6_final)
+# factor levels
+# rename the factor levels
+il6_final$channel_learned <- factor(
+    il6_final$channel_learned,
+    levels = c(
+        "nuclei",
+        "Mito",
+        "ER",
+        "gasdermin",
+        "PM",
+        "other"
+    ),
+    labels = c(
+        "Nuclei",
+        "Mito",
+        "ER",
+        "GasderminD",
+        "AGP",
+        "Other"
+    )
+)
+il6_final$feature_group <- factor(
+    il6_final$feature_group,
+    levels = c(
+        "AreaShape",
+        "Correlation",
+        "Granularity",
+        "Intensity",
+        "Location",
+        "Neighbors",
+        "RadialDistribution",
+        "Texture"
+    )
+)
+# reverese the order of the factor levels
+il6_final$feature_group <- factor(
+    il6_final$feature_group,
+    levels = rev(levels(il6_final$feature_group))
+)
+
+il6_final_plot <- plot_coeffs(il6_final, cytokine, shuffle)
+# output
+coef_gg_file <- file.path(paste0(output_path,"/","top_abs_val_coefficients_enet.pdf"))
+# set plot size
+width <- 8.5
+height <- 4
+options(repr.plot.width=width, repr.plot.height=height)
+ggsave(coef_gg_file, il6_final_plot, width=width, height=height, dpi=500)
+il6_final_plot
+
+
+cytokine <- 'CCL4 [NSU]'
+shuffle <- 'final'
+filename <- nested_list$filename[which(nested_list$cytokine == cytokine & nested_list$shuffle == shuffle)]
+# set to string
+filename <- filename[[1]]
+CCL4_final <- read.csv(filename, header = TRUE, sep = ",", stringsAsFactors = FALSE)
+
+CCL4_final <- process_subset_data(CCL4_final)
+head(CCL4_final)
+# factor levels
+# rename the factor levels
+CCL4_final$channel_learned <- factor(
+    CCL4_final$channel_learned,
+    levels = c(
+        "nuclei",
+        "Mito",
+        "ER",
+        "gasdermin",
+        "PM",
+        "other"
+    ),
+    labels = c(
+        "Nuclei",
+        "Mito",
+        "ER",
+        "GasderminD",
+        "AGP",
+        "Other"
+    )
+)
+CCL4_final$feature_group <- factor(
+    CCL4_final$feature_group,
+    levels = c(
+        "AreaShape",
+        "Correlation",
+        "Granularity",
+        "Intensity",
+        "Location",
+        "Neighbors",
+        "RadialDistribution",
+        "Texture"
+    )
+)
+# reverese the order of the factor levels
+CCL4_final$feature_group <- factor(
+    CCL4_final$feature_group,
+    levels = rev(levels(CCL4_final$feature_group))
+)
+
+CCL4_final_plot <- plot_coeffs(CCL4_final, cytokine, shuffle)
+# output
+coef_gg_file <- file.path(paste0(output_path,"/","top_abs_val_coefficients_enet.pdf"))
+# set plot size
+width <- 8.5
+height <- 4
+options(repr.plot.width=width, repr.plot.height=height)
+ggsave(coef_gg_file, CCL4_final_plot, width=width, height=height, dpi=500)
+CCL4_final_plot
```

Comment:
It seems to me like you did the same steps for the cytokines in cytokines_list. If this is the case, can you iterate through those strings to do all of this code?

Reply:
No because I need individual plot objects. I could write a function though